checker/src/main/java/org/checkerframework/checker/interning/InterningAnnotatedTypeFactory.java

```diff
@@ -65,6 +72,62 @@ public InterningAnnotatedTypeFactory(BaseTypeChecker checker) {
         this.postInit();
     }
 
+    @Override
+    protected DefaultQualifierForUseTypeAnnotator createDefaultForUseTypeAnnotator() {
+        return new InterningDefaultQualifierForUseTypeAnnotator(this);
+    }
+
+    /**
+     * Does not add defaults for type uses on constructor results. Constructor results should be
+     * {@code @UnknownInterned} by default.
+     */
+    static class InterningDefaultQualifierForUseTypeAnnotator
+            extends DefaultQualifierForUseTypeAnnotator {
+
+        public InterningDefaultQualifierForUseTypeAnnotator(AnnotatedTypeFactory typeFactory) {
+            super(typeFactory);
+        }
+
+        @Override
+        public Void visitExecutable(AnnotatedExecutableType type, Void p) {
+            MethodSymbol methodElt = (MethodSymbol) type.getElement();
+
+            if (methodElt == null
+                    || methodElt.getKind() != ElementKind.CONSTRUCTOR
+                    || methodElt.owner.getKind() == ElementKind.ENUM) {
+                // Annotate method returns, not constructors.
+                scan(type.getReturnType(), p);
+            }
+            if (type.getReceiverType() != null
+                    && typeFactory.getDeclAnnotation(type.getElement(), InternMethod.class)
```

Comment:
You could use `methodElt` again instead of calling `type.getElement()`.

Reply:
A comment would be useful to understand what should happen with methods that have the `InternMethod` annotation.